packages/backend-data/src/types.ts

```diff
@@ -147,3 +157,78 @@ export type AmplifyDataError =
+ */
+export type DataLoggingOptions = true | DataLogConfig;
+
+/**
+ * Customizable logging configuration when writing GraphQL operations and tracing to Amazon CloudWatch for an AWS AppSync GraphQL API.
+ *
+ * **WARNING**: Verbose logging will log the full incoming query including user parameters.
+ * Sensitive information may be exposed in CloudWatch logs. Ensure that your IAM policies only grant access to authorized users.
+ *
+ * For information on AppSync's LogConfig, refer to https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-appsync-graphqlapi-logconfig.html.
+ * For information on RetentionDays, refer to https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_logs.RetentionDays.html.
+ * @default excludeVerboseContent: true, fieldLogLevel: 'none', retention: '1 week'
+ */
+export type DataLogConfig = {
+  /**
+   * The number of days log events are kept in CloudWatch Logs.
+   * @default RetentionDays.ONE_WEEK
+   * @see https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_logs.RetentionDays.html
+   */
+  retention?: LogRetention;
+
+  /**
+   * When set to `true`, excludes verbose information from the logs, such as:
+   * - GraphQL Query
+   * - Request Headers
+   * - Response Headers
+   * - Context
+   * - Evaluated Mapping Templates
+   *
+   * This setting applies regardless of the specified logging level.
+   *
+   * **WARNING**: Verbose logging will log the full incoming query including user parameters.
+   * Sensitive information may be exposed in CloudWatch logs. Ensure that your IAM policies only grant access to authorized users.
+   * @default true
+   */
+  excludeVerboseContent?: boolean;
+
+  /**
+   * The field logging level. Values can be `'none'`, `'error'`, `'info'`, `'debug'`, or `'all'`.
+   *
+   * - **'none'**: No field-level logs are captured.
+   * - **'error'**: Logs the following information only for the fields that are in the error category:
+   *   - The error section in the server response.
+   *   - Field-level errors.
+   *   - The generated request/response functions that got resolved for error fields.
+   * - **'info'**: Logs the following information only for the fields that are in the info and error categories:
+   *   - Info-level messages.
+   *   - The user messages sent through `$util.log.info` and `console.log`.
+   *   - Field-level tracing and mapping logs are not shown.
+   * - **'debug'**: Logs the following information only for the fields that are in the debug, info, and error categories:
+   *   - Debug-level messages.
+   *   - The user messages sent through `$util.log.info`, `$util.log.debug`, `console.log`, and `console.debug`.
+   *   - Field-level tracing and mapping logs are not shown.
+   * - **'all'**: The following information is logged for all fields in the query:
+   *   - Field-level tracing information.
+   *   - The generated request/response functions that were resolved for each field.
+   * @default 'none'
+   */
+  level?: DataLogLevel;
```

Comment:
Should this be name `fieldLogLevel` just as you described in the default example in the docs above?

Reply:
Thanks for catching the discrepancy in the doc. `fieldLogLevel` is used in the cdk. But I think `level` as the API is more straightforward and aligns with what we already have in `defineFunction`. What do you think?

Reply:
Does AppSync have other log levels ? for example `modelLogLevel` ?
If it does have it or have potential to have it in the future then - would we add second "level property" here or use this `level` to feed both?

Reply:
I don't find any other log level besides `fieldLogLevel`. I'm confirm if AppSync has plan to add other log levels.

Reply:
But thanks for the catch. I think it's better to name it `fieldLogLevel` for clarity and to differentiate in case AppSync has plan later to add other log level